helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -828,12 +830,17 @@ public void onCustomizedStateRootChange(String instanceName, List<String> custom
     }
 
     // TODO: remove the synchronization here once we move this update into dataCache.
-    synchronized (_lastSeenCustomizedStateTypes) {
-      Set<String> lastSeenCustomizedStateTypes = _lastSeenCustomizedStateTypes.get();
+    synchronized (_lastSeenCustomizedStateTypesMap) {
+      Map<String, Set<String>> lastSeenCustomizedStateTypesMap =
+          _lastSeenCustomizedStateTypesMap.get();
+      Set<String> lastSeenCustomizedStateTypes = Collections.emptySet();
+      if (lastSeenCustomizedStateTypesMap != null && lastSeenCustomizedStateTypesMap
+          .containsKey(instanceName)) {
+        lastSeenCustomizedStateTypes = lastSeenCustomizedStateTypesMap.get(instanceName);
+      }
       for (String customizedState : customizedStateTypes) {
         try {
-          if (lastSeenCustomizedStateTypes == null || !lastSeenCustomizedStateTypes
-              .contains(customizedState)) {
+          if (!lastSeenCustomizedStateTypes.contains(customizedState)) {
             manager.addCustomizedStateChangeListener(this, instanceName, customizedState);
             logger.info(
```

Comment:
nit, if addCustomizedStateChangeListener already logs inside, we don't need this info log. Please confirm.

Reply:
In ZkHelixManager, there's log for addListener, but in the controller, all the operations of adding listeners have this log, so I think it's better to keep consistent.

Reply:
The only additional information we get by printing this log is manager.getInstanceName(), I guess. I don't have a strong preference. But do whatever the other similar code does might cause redundant code. Please only keep this log if you think the extra helixmanager name can help us to debug in the future.

Reply:
I think it can still provide some information, so keep it for now.